crates/chia_py_streamable_macro/src/lib.rs

```diff
@@ -174,45 +189,91 @@ pub fn py_streamable_macro(input: proc_macro::TokenStream) -> proc_macro::TokenS
+                use pyo3::IntoPy;
+                use pyo3::type_object::PyTypeInfo;
+                use std::borrow::Borrow;
+
                 if !blob.is_c_contiguous() {
                     panic!("from_bytes() must be called with a contiguous buffer");
                 }
                 let slice = unsafe {
                     std::slice::from_raw_parts(blob.buf_ptr() as *const u8, blob.len_bytes())
                 };
-                <Self as #crate_name::Streamable>::from_bytes(slice).map_err(|e| <#crate_name::chia_error::Error as Into<pyo3::PyErr>>::into(e))
+                let rust_obj = <Self as #crate_name::Streamable>::from_bytes(slice)?;
+
+                // Check if python class is different from rust class (in case of child classes)
+                // if so call the python class's conversion code
+
+                let cls_type = cls.borrow().as_ref();
+                if <Self as PyTypeInfo>::is_exact_type_of_bound(cls_type) {
+                    Ok(rust_obj.into_py(py))
+                } else {
+                    let instance = cls.call_method1("from_parent", (rust_obj.into_py(py),))?;
+                    Ok(instance.into_py(py))
+                }
             }
 
-            #[staticmethod]
+            #[classmethod]
             #[pyo3(name = "from_bytes_unchecked")]
-            pub fn py_from_bytes_unchecked(blob: pyo3::buffer::PyBuffer<u8>) -> pyo3::PyResult<Self> {
+            pub fn py_from_bytes_unchecked(cls: &pyo3::Bound<'_, pyo3::types::PyType>, py: pyo3::Python<'_>, blob: pyo3::buffer::PyBuffer<u8>) -> pyo3::PyResult<pyo3::PyObject> {
+                use pyo3::prelude::PyAnyMethods;
+                use pyo3::IntoPy;
+                use pyo3::type_object::PyTypeInfo;
+                use std::borrow::Borrow;
                 if !blob.is_c_contiguous() {
                     panic!("from_bytes_unchecked() must be called with a contiguous buffer");
                 }
                 let slice = unsafe {
                     std::slice::from_raw_parts(blob.buf_ptr() as *const u8, blob.len_bytes())
                 };
-                <Self as #crate_name::Streamable>::from_bytes_unchecked(slice).map_err(|e| <#crate_name::chia_error::Error as Into<pyo3::PyErr>>::into(e))
+                let rust_obj = <Self as #crate_name::Streamable>::from_bytes_unchecked(slice).map_err(|e| <#crate_name::chia_error::Error as Into<pyo3::PyErr>>::into(e))?;
+
+                // Check if python class is different from rust class (in case of child classes)
+                // if so call the python class's conversion code
+
+                let cls_type = cls.borrow().as_ref();
+                if <Self as PyTypeInfo>::is_exact_type_of_bound(cls_type) {
+                    Ok(rust_obj.into_py(py))
+                } else {
+                    let instance = cls.call_method1("from_parent", (rust_obj.into_py(py),))?;
+                    Ok(instance.into_py(py))
+                }
```

Comment:
is there any chance this snippet could be factored out into a function? or something else to address the apparent copy-pasted code